### PR TITLE
feat(frontend): App stepper

### PR DESCRIPTION
**Pull request by fatonramadani**
https://user-images.githubusercontent.com/456655/236400739-bf7f8de1-b23a-4916-b29a-a097b259f7c2.mov

https://user-images.githubusercontent.com/456655/236400751-87687157-2856-4cbd-9e3c-d091d2d386c7.mov

### CODE REVIEW

File: frontend/src/lib/components/apps/components/layout/AppStepper.svelte [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+		connectingInput,
+		mode
+	} = getContext<AppViewerContext>('AppViewerContext')
+
+	let selected: string = tabs[0]
+	let tabHeight: number = 0
+	let footerHeight: number = 0
+	let runnableComponent: RunnableComponent
+	let selectedIndex = tabs?.indexOf(selected) ?? -1
+	$: maxReachedIndex = -1
```
Comment (rubenfiszel) [inline frontend/src/lib/components/apps/components/layout/AppStepper.svelte]: ???

File: frontend/src/lib/components/apps/components/layout/AppStepper.svelte [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+	export let id: string
+	export let componentContainerHeight: number
+	export let tabs: string[]
+	export let customCss: ComponentCustomCSS<'steppercomponent'> | undefined = undefined
+	export let render: boolean
+	export let recomputeIds: string[] | undefined = undefined
+	export let extraQueryParams: Record<string, any> = {}
+	export let componentInput: AppInput | undefined
+
+	$: statusByStep = [] as Array<'success' | 'error' | 'pending'>
```
Comment (rubenfiszel) [inline frontend/src/lib/components/apps/components/layout/AppStepper.svelte]: ???

File: frontend/src/lib/components/apps/editor/ComponentHeader.svelte [hunk 3]
```diff
@@ -108,7 +108,7 @@
 	</div>
 {/if}
 
-{#if error}
+{#if error && component.type !== 'steppercomponent'}
 	{@const json = JSON.parse(JSON.stringify(error))}
```
Comment (rubenfiszel) [inline frontend/src/lib/components/apps/editor/ComponentHeader.svelte]: why do we do this ?

File: frontend/src/lib/components/apps/editor/component/default-codes.ts [hunk 4]
```diff
@@ -46,6 +46,18 @@ export const DEFAULT_CODES: Partial<
 export async function main(db: Resource<"postgresql"> = "$res:f/examples/demodb") {
 	const query = await pgSql(db)\`SELECT * FROM demo;\`;
 	return query.rows;
+}`
+	},
+	steppercomponent: {
```
Comment (rubenfiszel) [inline frontend/src/lib/components/apps/editor/component/default-codes.ts:51]: need the equivalent in python

Comment (rubenfiszel) [reply frontend/src/lib/components/apps/editor/component/default-codes.ts:51]: the structure is not super intuitive also need formatting
```
  // if page == 0 {
  //  if (page0Invalid) throw Error("first step invalid")
  // } else if ...
  if (final) {
     submitFullForm()
  }
```

Also we should call it `lastPage`

**Comment by rubenfiszel** [comment]
I think doing the pre connection is pretty important

File: frontend/src/lib/components/apps/components/layout/AppStepper.svelte (hunk 2 above)
Comment (fatonramadani) [reply frontend/src/lib/components/apps/components/layout/AppStepper.svelte]: Removed $:

File: frontend/src/lib/components/apps/editor/component/default-codes.ts (hunk 4 above)
Comment (fatonramadani) [reply frontend/src/lib/components/apps/editor/component/default-codes.ts:51]: Done!

File: frontend/src/lib/components/apps/editor/inlineScriptsPanel/InlineScriptEditor.svelte [hunk 5]
```diff
@@ -72,11 +73,39 @@
 	const dispatch = createEventDispatcher()
 	let runLoading = false
 
+	function preConnect(newFields) {
```
Comment (fatonramadani) [inline frontend/src/lib/components/apps/editor/inlineScriptsPanel/InlineScriptEditor.svelte:76]: @rubenfiszel  the pre-connect logic

**Pull request merged**